### PR TITLE
[ADD] delivery_free_fee_removal: avoid create delivery line if price is zero

### PR DESCRIPTION
**Pull request by cubells**
cc @Tecnativa

### CODE REVIEW

File: delivery_free_fee_removal/README.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/delivery-carrier-11-0/delivery-carrier-11-0-delivery_free_fee_removal
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/99/11.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows to avoid create a shipping line in sale order if its price
```
Comment (pedrobaeza) [inline delivery_free_fee_removal/README.rst:28]: s/shipping/delivery

File: delivery_free_fee_removal/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+from odoo import models
+from odoo.tools.float_utils import float_is_zero
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _create_delivery_line(self, carrier, price_unit):
+        sale_line_object = self.env['sale.order.line']
```
Comment (pedrobaeza) [inline delivery_free_fee_removal/models/sale_order.py]: You can remove this variable and return directly `self.env['sale.order.line']`

File: delivery_free_fee_removal/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        rounding = self.currency_id.rounding
+        if float_is_zero(price_unit, precision_rounding=rounding):
+            return sale_line_object
+        return super(SaleOrder, self)._create_delivery_line(carrier,
```
Comment (pedrobaeza) [inline delivery_free_fee_removal/models/sale_order.py]: You can put it in one line:
```python
return super()._create_delivery_line(carrier, price_unit)
```

**Review by pedrobaeza — APPROVED** [review]
Some nitpickings, but it's overall OK. Having more test lines than the code itself is a great signal I think!

**Comment by cubells** [comment]
@pedrobaeza done.

**Review by tarteo — APPROVED** [review]

**Pull request merged**